Demo cn0398

adrimbarean opened the pull request.

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,443 @@
+    /* Initialize the system */
+    InitSystem();
+
+    /* Open Accelerometer */
```
no accelerometer sensor inside cn0398

stefpopa replied:
As a general remark, please run astyle on your files

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/RTE/Sensors/adi_adxl362_config.h:
```diff
@@ -0,0 +1,79 @@
+#define ADI_CFG_ADXL362_RATE  (adi_sensor_swpack::Accelerometer::AXL_ODR_100)
+
+#endif  /* ADI_ADXL362_CFG_H */
+/* @} */
```
file adi_adxl362_config.h is not required for cn0398

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,443 @@
+        Trap();
+    }
+
+	/* Publish ADXL362 accelerometer data using MQTT */
```
adxl362 not present

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.h:
```diff
@@ -0,0 +1,102 @@
+
+/********************************* Wi-Fi Configuration *********************************/
+/* SSID of the access point. */
+uint8_t aWifiSSID[] = "AnalogDevices";
```
leave blank and add a comment (don't give away local SSID and passwords to customers)

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/Timer.c:
```diff
@@ -0,0 +1,117 @@
+void timer_start (void)
+{
+
+//    if(adi_pwr_Init()!= ADI_PWR_SUCCESS)
```
remove commented code

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/adi_cn0398.cpp:
```diff
@@ -0,0 +1,626 @@
+		calibration_ph[0][1] =  volt;
+		set_digital_output(P2, false);
+
+//		sprintf((char *)ui8S,"\tCalibration solution1 pH = %.3f with sensor voltage of %fV\n", calibration_ph[0][0], volt);
```
remove commented code

mirceacaprioru replied:
or uncomment in this case

mirceacaprioru replied:
calibration must be available over MQTT also

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/adi_cn0398.cpp:
```diff
@@ -0,0 +1,626 @@
+		calibration_ph[1][1] =  volt;
+		set_digital_output(P2, false);
+
+//		sprintf((char *)ui8S,"\tCalibration solution2 pH = %.3f with sensor voltage of %fV\n\r", calibration_ph[1][0], volt);
```
calibration is required for cn0398

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/adi_cn0398.cpp:
```diff
@@ -0,0 +1,626 @@
+	{
+		uint8_t ui8S[200];
+
+//		sprintf((char *)ui8S,"Temperature = %f°C\n\r pH = %f\n\r Moisture = %f%c\n\r", temperature, pH, moisture, 37);
```
display data must be tailored to use MQTT for this example

mirceacaprioru commented on projects/ADuCM3029_demo_cn0398/src/adi_cn0398.cpp:
```diff
@@ -0,0 +1,626 @@
+//		AppPrintf("\n");
+
+	}
+	void CN0398::calibrate_ph(void)
```
To provide full functionality calibration is necessary

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+int main(int argc, char *argv[])
+{
+    CN0398         	cn0398;
+//    SoilMoisture	*pCn = &cn0398;
```
Please remove commented code

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+													 .nLinkID = ADI_WIFI_CONNECTION_ID,
+													 .nQos = ADI_WIFI_MQTT_PUBLISER_QOS};
+
+//uint8_t convFlag;
```
remove commented code

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+static void InitSystem(void);
+static void Trap(void);
+static void RepairConnection(void);
+
```
nitpick: you have one extra line here :)

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+{
+	switch (nEvent)
+		{
+			case CMD_NONE:
```
make sure that you respect the Linux coding style indentations, for example this line should be:
`switch (nEvent) {`
take a look at: https://www.kernel.org/doc/html/v4.10/process/coding-style.html

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+				adi_wifi_ParseSubscriberData();
+				break;
+			case CMD_HW_ERROR:
+			{
```
you do not need curly braces `{ ` inside a case

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+
+    /* Start the ping timer to ping the MQTT broker before the keep alive expires */
+    if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) == ADI_WIFI_SUCCESS)
+    {
```
the opening brace should be put last on the line: `if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) == ADI_WIFI_SUCCESS) {`

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+                    {
+        		    	DEBUG_MESSAGE("Failed to start ping timer\n");
+
+                    }
```
Do not unnecessarily use braces where a single statement will do: `if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) != ADI_WIFI_SUCCESS)
                DEBUG_MESSAGE("Failed to start ping timer\n");`

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+
+	if(adi_wifi_radio_GetConnectionStatus(&nConnectionStatus) == ADI_WIFI_SUCCESS)
+	{
+
```
nitpick: one extra line

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+    timer_start();
+
+    DEBUG_MESSAGE("Starting MQTT publisher Wi-Fi ADXL362 accelerometer demo application\r\n");
+
```
extra line

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+	if(adi_wifi_radio_GetConnectionStatus(&nConnectionStatus) == ADI_WIFI_SUCCESS)
+	{
+
+		switch(nConnectionStatus)
```
I would use an enum to define the different cases

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+//uint8_t convFlag;
+uint8_t m_spi_memory[ADI_SPI_MEMORY_SIZE];
+
+//extern float temperature, pH, moisture;
```
please remove commented code

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+	 * Initialize managed drivers and/or services that have been added to 
+	 * the project.
+	 * @return zero on success 
+	 */
```
I think this comment should be  before the definition of a function, not inside one

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+				if(adi_wifi_radio_MQTTPublish(&gPublishConfig) == ADI_WIFI_FAILURE)
+				{
+					nConnectionFailFlag = 1u;
+				}
```
Do not unnecessarily use braces where a single statement will do.

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,403 @@
+				{
+					nConnectionFailFlag = 1u;
+				}
+
```
Do not unnecessarily use braces where a single statement will do.

stefpopa commented:
Just as a general remark, you should add a changelog, a few bullet points with what has changed

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+													 .pTopic = aMQTTTopicName,
+													 .nLinkID = ADI_WIFI_CONNECTION_ID,
+													 .nQos = ADI_WIFI_MQTT_PUBLISER_QOS};
+
```
alignment looks strange here, you are too far to the right; the limit on the length of lines is 80 columns and this is a strongly preferred limit.

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+/* Local Wi-Fi functions */
+static void            InitWiFiConnection(void);
+static void            adi_wifi_AplicationCallback(void * pCBParam, uint32_t nEvent, void * pArg);
+static void			   MQTTPublishCN0398(CN0398 pCn);
```
nitpick: alignment looks strange

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+	 * Initialize managed drivers and/or services that have been added to 
+	 * the project.
+	 * @return zero on success 
+	 */
```
alignment looks strange

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+	MQTTPublishCN0398(cn0398);
+
+	return 0;
+}
```
this section of code is not aligned

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+    }
+}
+
+/*!
```
the whole function's indentation is very strange; make sure you run astyle

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+			/*The  Station is connected to an AP and its IP is obtained, so try to connect to broker. */
+			case 2:
+				/* Send MQTT "CONNECT" message to broker */
+				if(adi_wifi_radio_MQTTConnect(&gMQTTConnect) != ADI_WIFI_SUCCESS)
```
case should be under the switch, you have 1 extra tab, it should be something like
```
switch (action) {
case KOBJ_ADD:
        return "add";
case KOBJ_REMOVE:
        return "remove";
case KOBJ_CHANGE:
        return "change";
default:
        return NULL;
}
```

take a look at: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/adi_cn0398.h:
```diff
@@ -0,0 +1,194 @@
+	float read_moisture();
+
+
+
```
you have a few extra blank lines here

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4:
```diff
@@ -0,0 +1,384 @@
+		}
+
+	return;
+}
```
case should be under the switch, you have a few extra tabs, it should be something like
```
switch (action) {
case KOBJ_ADD:
        return "add";
case KOBJ_REMOVE:
        return "remove";
case KOBJ_CHANGE:
        return "change";
default:
        return NULL;
}
```

take a look at: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces

adrimbarean commented:
changelog v1 -> v2:
* changed comments to correctly mention cn0398 instead of adxl362
* removed adi_adxl362_config.h file as it is not part of this demo
* changed configuration data regarding WiFi connection (SSID and password) with **** (neutral value that does not compromise security)
* removed commented code
* added calibration feature with user interface
* fixed data output to be readable ascii rather than raw data from adc reads
* added a pointer directly to received message from MQTT, discarding MQTT header of 4 bytes and topic 
name
* fix whitespace and indentation
* fix coding style (brackets and indentation)

stefpopa commented on projects/.metadata/.plugins/org.eclipse.core.resources/.history/16/90fdfef8ef1d00181353d905c66b77c4:
```diff
@@ -0,0 +1,79 @@
+/*!
```
as a general remark: remove the unnecessary history files

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+};
+
+extern volatile uint8_t ui8S[200];
+uint8_t *payload_ptr = (uint8_t*)(ui8S + 4 + sizeof(aMQTTTopic) - 1);
```
use a #define instead of the "magic number" '4'

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+		rec_flag = false;
+		if(payload_ptr[0] == 'y' || payload_ptr[0] == 'Y')
+			pCn.use_nernst = false;
+	}
```
remember the if/else special case for brackets:
```
if (condition) {
        do_this();
        do_that();
} else {
        otherwise();
}
```
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+					DEBUG_MESSAGE("Failed to ping broker\n");
+					nConnectionFailFlag = 1;
+				}
+			}
```
same remark as above

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+
+	if(adi_wifi_radio_GetConnectionStatus(&nConnectionStatus)
+	    == ADI_WIFI_SUCCESS)
+		switch(nConnectionStatus) {
```
I would rather place only `ADI_WIFI_SUCCESS` on the new line and leave '==' on the previous line

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+			if(adi_wifi_radio_EstablishTCPConnection(&gTCPConnect)
+			    == ADI_WIFI_SUCCESS) {
+				if(adi_wifi_radio_MQTTConnect(&gMQTTConnect)
+				    != ADI_WIFI_SUCCESS)
```
same remark as above

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+		case 5:
+			if(adi_wifi_radio_ConnectToAP(aWifiSSID, aWifiPassword,  NULL)
+			    == ADI_WIFI_SUCCESS) {
+				if(adi_wifi_radio_EstablishTCPConnection(&gTCPConnect)
```
same remark as above

stefpopa commented on projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp:
```diff
@@ -0,0 +1,430 @@
+			if(adi_wifi_radio_ConnectToAP(aWifiSSID, aWifiPassword,  NULL)
+			    == ADI_WIFI_SUCCESS) {
+				if(adi_wifi_radio_EstablishTCPConnection(&gTCPConnect)
+				    == ADI_WIFI_SUCCESS) {
```
same remark

adrimbarean commented:
changelog v2 -> v3:
* enforce 80 column limit
* moved == to be on the first line before line feed
* added error message in wifi callback function
* corrected if-else brackets {}
* changed comments from // to /**/ format to match whole code
* removed commented code
* add comment that explains commented code and when to uncomment it
* remove history files
* remove unused variables
* used define instead of magic number
